src/OrchardCore.Modules/Orchard.Settings/AdminMenu.cs

```diff
@@ -28,6 +28,15 @@ public void BuildNavigation(string name, NavigationBuilder builder)
                     )
                 )
             );
+
```

Comment:
This might be a better place to check for user permission before adding the button.